src/keys.rs

```diff
@@ -281,77 +279,72 @@ impl IssuanceAuthorizingKey {
             .map(|esk| esk.sk().into())
     }
 
-    /// Derives the RedPallas signing key from isk. Internal use only, does not enforce all constraints.
-    fn derive_inner(&self) -> pallas::Scalar {
-        to_scalar(PrfExpand::ZsaIsk.expand(&self.0))
+    /// Sign the provided message using the `IssuanceAuthorizingKey`.
+    /// Only supports signing of messages longer than 32 bytes, using prehashing, as described in [BIP 340][bip340]
+    ///
+    /// [bip340]: https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#messages-of-arbitrary-size
+    pub fn sign(&self, msg: &[u8]) -> Signature {
+        assert!(msg.len() >= 32);
+        schnorr::SigningKey::from(self.0).sign(msg)
```

Comment:
Is the `assert!` okay then? That could cause `try_sign` to panic, for a reason different than a signature error though.

We could define a new Error enum that keeps `MessageTooShort` as one variant, and have the other variant include the `schnorr::Error` values as a field, like below:

```
enum Error {
    MessageTooShort,
    SigError(schnorr::Error),
}
```

Reply:
good idea,
first check if `schnorr::Error` has something similar to what we need. if not, let's create this enum.

Reply:
The subsequent discussion we had made this unnecessary. We now use the `sign_prehash` function from the `PrehashSigner` trait, which raises an error if the length of the message being signed is not exactly 32 bytes. This is what we need since we are only signing Issue Bundle SIGHASH values.